management/commands/import_plos2EA.py

```diff
@@ -0,0 +1,495 @@
+import traceback
+import sys
+
+PRESS_ID = 1
+REPO_ID = 1
+COI_ID = 1
+DATA_ID = 1
+CCBY_ID = 1
+REPO_NAME = 'EarthArXiv (Dev)'
+SUCCESS_TO_MAIL = ['mahjabeen.yucekul@ucop.edu', 'alainna.wrigley@ucop.edu']
+FAILURE_TO_MAIL = ['mahjabeen.yucekul@ucop.edu', 'alainna.wrigley@ucop.edu']
+#SUCCESS_TO_MAIL = ['mahjabeen.yucekul@ucop.edu','eartharxiv_notifications@plos.org']
+#FAILURE_TO_MAIL = ['mahjabeen.yucekul@ucop.edu','help@escholarship.org']
+class Command(BaseCommand): 
+    """
+    Pulls data from COS and adds to DB.
+    """
+
+    help = "Imports PLOS data from ftps to EA"
+
+    def handle(self, *args, **options):
+        print("import plos to EA data")
+        if self.check_prereq() == False:
+            return
+        #self.testEmail()
+        print(django_settings.IMPORT_TOKEN)
+
+        w = Worker()
+        w.processAll()
+
+    def check_prereq(self):
+        print("checking prereq")
+        if django_settings.IMPORT_TOKEN is None:
+            print("add Import token to settings to proceed")
+            return False
+
+        # get repository
+        global REPO_ID
+        global PRESS_ID
+        global COI_ID
+        global DATA_ID
+        global CCBY_ID
+
+        # TBD - fix this check
+        repo = repository_models.Repository.objects.get(name=REPO_NAME)
+        REPO_ID = repo.id
+        PRESS_ID = repo.press_id
+        COI_ID = repository_models.RepositoryField.objects.get_or_create(name='Conflict of interest statement', input_type='text', repository_id = REPO_ID, defaults={'required':0, 'order':1, 'display':0 })[0].id
+        DATA_ID = repository_models.RepositoryField.objects.get_or_create(name='Data Availability (Reason not available)', input_type='text', repository_id = REPO_ID, defaults={'required':0, 'order':2, 'display':0 })[0].id
+        CCBY_ID = submission_models.Licence.objects.get(name='CC BY Attribution 4.0 International',press_id = PRESS_ID).id
+
+        print("REPO_ID" + str(REPO_ID))
+        print("PRESS_ID" + str(PRESS_ID))
+        print("COI_ID" + str(COI_ID))
+        print("DATA_ID" + str(DATA_ID))
+        print("CCBY_ID" + str(CCBY_ID))
+
+        return True
+
+    def testEmail(self):
+        send_mail(
+            'Test Subject',
+            'Test message.',
+            django_settings.DEFAULT_FROM_EMAIL ,
+            SUCCESS_TO_MAIL,
+            fail_silently=False,
+        )
+
+
+###################
+#### Main Worker ##
+###################
+#############################
+class Worker:
+    workingDir = '/apps/eschol/import/plos/working'
+    doneDir = '/apps/eschol/import/plos/done'
+    errorDir = '/apps/eschol/import/plos/error'
+    remote = 'ftps.cdlib.org'
+    section = 'plos'
+    importkey = django_settings.IMPORT_TOKEN
+    filesFound = []
+    def __init__(self):
+        print("created command")
+        # get staged files
+        self.getStagedFiles()
+        # look over the files and process
+
+    def getStagedFiles(self):

```

Comment:
This places trust in the stability of the network connection, to process the files with the FTP session still going. I think this is fine, but we may discover that it's not, and then we'll need to separate the fetch stage and the process stage. Just an observation, not a suggestion to change anything.

Reply:
There is no processing when files are grabbed from FTP. The functions are decoupled. Files are downloaded, removed from FTP and then processed from the working directory.